docs/C_STYLE_GUIDE.md

```diff
@@ -0,0 +1,112 @@
+# C Style Guide
+
+This guide describes the style intended for use with the C track exercise 
+example and test files.
+
+The rules within this guide are in addition to, or to call out, those in the 
+[Linux kernel coding style guide](https://www.kernel.org/doc/html/latest/process/coding-style.html).
+
+## Names
+
+In general, the names of types, variables, structures and files should use 
+snake case with all letters in lower case, where underscores `_` are used to 
+separate name components.
+
+```c
+void some_foo_function(char * input_string, int input_string_length);
+```
+
+## Enums
+
+Enum names should usually be singluar rather than plural.
```

Comment:
`singluar` => `singular`

Reply:
Dammit! I type too fast and then read only what I expect to be there when I check it. Thanks for catching that.